codice/ddf
# DDF-4982 Improve Search Form and Map Action UX

#### What does this PR do?
- Refactors search forms to reduce duplicated code
- Fixes UX issues with search and result forms
	- Switching tabs no longer refreshes the entire view (including menu bar)
	- Proper messages are shown when no forms are found.
- Fixes UX issues with map actions
	- Actions acting on selected results now identify as such

#### Who is reviewing it? 
@djblue 
@samuelechu 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@adimka
@bdeining

#### How should this be tested?
Map Actions
1. Interact with the map and click map actions to ensure they still work as expected.

Search Forms
1. Download [forms.zip](https://github.com/codice/ddf/files/3308001/forms.zip), unzip, and place contents in `$DDF_HOME/etc/forms/`.
2. In Karaf console run `forms:load` to load system templates.
3. Test
a. Creating a search form
b. Sharing a search form (and opening on another user)
c. Opening system search forms
d. Querying with search forms


#### What are the relevant tickets?
Fixes: #4982 

#### Screenshots
##### Search Forms
*Before*
![DDF-4982](https://user-images.githubusercontent.com/7236299/59776777-3c740000-9268-11e9-98b1-b1b496de5724.gif)
*After*
![SearchFormsAfter](https://user-images.githubusercontent.com/7236299/59777449-86a9b100-9269-11e9-99cc-f2d19430afcf.gif)

##### Map Actions
*Before*
![image](https://user-images.githubusercontent.com/7236299/59777018-ac828600-9268-11e9-899b-4dd453ee8732.png)

*After*
![image](https://user-images.githubusercontent.com/7236299/59777255-303c7280-9269-11e9-87d8-a255a612f6ab.png)


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
